File: mobile/src/main/java/ro/antiprotv/sugar/db/MeasureUnit.java
package ro.antiprotv.sugar.db;

public enum MeasureUnit {
    KGS(1), BOXES(2), BOTTLES(3), PIECES(4), PACKS(5), OTHER(99);

    int code;

    MeasureUnit(int code) {
        this.code = code;
    }

    public static int getCodeFromString(String measureUnit) {
        int code = 99;
        for (MeasureUnit m :
                MeasureUnit.values()) {
            if (m.toString().equals(measureUnit)) {
                code = m.code;
                break;
            }
        }
        return code;
    }

    public static String fromCode(int code){
        MeasureUnit mu = MeasureUnit.OTHER;
        for (MeasureUnit m :
                MeasureUnit.values()) {
            if (m.code == code) {
                mu = m;
                break;
            }
        }
        return mu.toString();
    }
}

File: mobile/src/main/java/ro/antiprotv/sugar/repository/db/MeasureUnit.kt
package ro.antiprotv.sugar.repository.db

enum class MeasureUnit(val code: Int, val resId: String) {
    KG(1, "item_unit_kg"),
    BOX(2, "item_unit_box"),
    BOTTLE(3, "item_unit_bottle"),
    PIECE(4, "item_unit_pc"),
    PACK(5, "item_unit_pack"),
    OTHER(99, "item_unit_other");

    companion object {
        fun fromString(measureUnitName: String): MeasureUnit {
            var measureUnit = OTHER
            for (m in values()) {
                if (m.toString() == measureUnitName) {
                    measureUnit = m
                    break
                }
            }
            return measureUnit
        }

        fun fromCode(code: Int): String {
            var mu = OTHER
            for (m in values()) {
                if (m.code == code) {
                    mu = m
                    break
                }
            }
            return mu.toString()
        }
    }

}